src/frequenz/sdk/timeseries/_grid_frequency.py

```diff
@@ -0,0 +1,157 @@
+from ..timeseries._base_types import Sample
+from ..timeseries._quantities import Frequency
+
+
+def create_request(component_id: int) -> "ComponentMetricRequest":  # type: ignore
+    """Create a request for grid frequency.
+
+    Args:
+        component_id: The component id to use for the request.
+
+    Returns:
+        A component metric request for grid frequency.
+    """
+    # Imported here to avoid a circular import.
+    # pylint: disable=import-outside-toplevel
+    from ..actor import ComponentMetricRequest
+
+    return ComponentMetricRequest(
+        "grid-frequency", component_id, ComponentMetricId.FREQUENCY, None
+    )
+
+
+class GridFrequency:
+    """Grid Frequency."""
+
+    def __init__(
+        self,
+        data_sourcing_request_sender: Sender["ComponentMetricRequest"],  # type: ignore
+        channel_registry: ChannelRegistry,
+        component: Component,
+    ):
+        """Initialize the grid frequency formula generator.
+
+        Args:
+            channel_registry: The channel registry to use for the grid frequency.
+            data_sourcing_request_sender: The sender to use for requests.
+            component: The component to use for the grid frequency receiver. If not
+                provided, the first component that is either a meter, inverter or EV
+                charger will be used.
+        """
+        self._request_sender = data_sourcing_request_sender
+        self._channel_registry = channel_registry
+        self._component = component
+        self._component_metric_request = create_request(component.component_id)
+
+        self._task: None | asyncio.Task[None] = None
+
+    @property
+    def component(self) -> Component:
+        """The component that is used for grid frequency.
+
+        Returns:
+            The component that is used for grid frequency.
+        """
+        return self._component
+
+    def new_receiver(self) -> Receiver[Sample[Frequency]]:
+        """Create a receiver for grid frequency.
+
+        Returns:
+            A receiver that will receive grid frequency samples.
+        """
+        receiver = self._channel_registry.new_receiver(
+            self._component_metric_request.get_channel_name()
+        )
+
+        if not self._task:
+            self._task = asyncio.create_task(self._send_request())
+        else:
+            logging.info("Grid frequency request already sent: %s", self._component)
+
+        return receiver
+
+    async def _send_request(self) -> None:
+        """Send the request for grid frequency."""
+        logging.info("Sending request for grid frequency: %s", self._component)
+        await self._request_sender.send(self._component_metric_request)
+        logging.info("Sent request for grid frequency: %s", self._component)
+
+    @staticmethod
+    def find_frequency_component() -> Component:
+        """Find the component that will be used for grid frequency.
```

Comment:
Maybe it would be good to explain the algorithm here? What does it try to find first? What happens if that doesn't exist,etc.?